### ISSUE TITLE
Incorrect MD032 for long list item with special characters

### ISSUE THREAD

**Issue by bryanforbes** [rvben/rumdl#188]
Starting with the following document using the default config (`rumdl --no-config check TEST.md`) and version 0.0.191:

```markdown
# Test

- **Token Dispatch (Phase 3.2)**: COMPLETE. Extracts tokens from both:
  1. Switch/case dispatcher statements (original Phase 3.2)
  2. Inline conditionals - if/else, bitwise checks (`&`, `|`), comparison (`==`, `!=`), ternary operators (`?:`), macros (`ISTOK`, `ISUNSET`), compound conditions (`&&`, `||`) (Phase 3.2.1)
     - 30 explicit tokens extracted, 23 dispatcher rules with embedded token
       references
```

Line 5 is correctly flagged as a violation of MD013, so I manually fixed it to the following (there's no code action to reflow the line, which may be another bug for another day):

```markdown
# Test

- **Token Dispatch (Phase 3.2)**: COMPLETE. Extracts tokens from both:
  1. Switch/case dispatcher statements (original Phase 3.2)
  2. Inline conditionals - if/else, bitwise checks (`&`, `|`), comparison (`==`,
     `!=`), ternary operators (`?:`), macros (`ISTOK`, `ISUNSET`),
     compound conditions (`&&`, `||`) (Phase 3.2.1)
     - 30 explicit tokens extracted, 23 dispatcher rules with embedded token
       references
```

Line 7 is now flagged as a violation of MD032. I've tried a few different combinations to try to determine how many of the special characters I can remove to silence MD032 and removing all of them seems to do the trick:

```markdown
# Test

- **Token Dispatch (Phase 3.2)**: COMPLETE. Extracts tokens from both:
  1. Switch/case dispatcher statements (original Phase 3.2)
  2. Inline conditionals - if/else, bitwise checks, comparison, ternary
     operators, macros (`ISTOK`, `ISUNSET`), compound conditions (Phase 3.2.1)
     - 30 explicit tokens extracted, 23 dispatcher rules with embedded token
       references
```

**Comment by rvben** [comment]
Thanks @bryanforbes for raising this!

There was some oversimplified logic that checked for tables in the line so that it wouldn't auto-reflow tables. Table detection is now smarter so it should show up now as a Quick Fix too. Can you check to see if all issues you encountered are fixed in 0.0.192?

**Comment by bryanforbes** [comment]
It looks like the major issue (flagging MD032) is fixed. However, while I was manually reflowing the line in question, I noticed that some other errors aren't being flagged now. Starting with the same starting document:

```markdown
# Test

- **Token Dispatch (Phase 3.2)**: COMPLETE. Extracts tokens from both:
  1. Switch/case dispatcher statements (original Phase 3.2)
  2. Inline conditionals - if/else, bitwise checks, comparison, ternary operators (`?:`), macros (`ISTOK`, `ISUNSET`), compound conditions
     (Phase 3.2.1)
     - 30 explicit tokens extracted, 23 dispatcher rules with embedded token
       references
```

I move to line 5, column 82, backspace over the space and hit enter to get this:

```markdown
# Test

- **Token Dispatch (Phase 3.2)**: COMPLETE. Extracts tokens from both:
  1. Switch/case dispatcher statements (original Phase 3.2)
  2. Inline conditionals - if/else, bitwise checks (`&`, `|`), comparison (`==`,
`!=`), ternary operators (`?:`), macros (`ISTOK`, `ISUNSET`), compound conditions (`&&`, `||`) (Phase 3.2.1)
     - 30 explicit tokens extracted, 23 dispatcher rules with embedded token
       references
```

I get the following errors (which seem correct):

```text
TEST.md:5:1: [MD032] List should be followed by blank line [*]
TEST.md:6:81: [MD013] Line length 108 exceeds 80 characters
TEST.md:7:1: [MD032] List should be preceded by blank line [*]
```

Then I start indenting line 6 and get some interesting results. Indenting by 1 space gives the same results as above. Indenting by 2 or 3 spaces gives the following results:

```text
TEST.md:5:1: [MD032] List should be followed by blank line [*]
TEST.md:6:81: [MD013] Line length 110 exceeds 80 characters
```

And then indenting by 4 spaces gives the following results:

```text
TEST.md:6:81: [MD013] Line length 112 exceeds 80 characters
```

These last two are a bit surprising because I would have figured there would be a rule to align the list text with the first character of the list text. I know that may be a separate issue, but I figured I'd bring it up since you're asking. The "reflow paragraph" code action also never shows up for the line, but that's probably also a new issue. Other than those things, it looks like this is fixed.

**Comment by rvben** [comment]
The MD032 rule was incorrectly flagging multi-line list items when they contained special characters like `||`, `&&`, `==`, etc. The root cause was two-fold:

1. **Pipe characters in inline code** (like `` `||` `` or `` `&&` ``) were being mistaken for table delimiters
2. **Block end line tracking** wasn't properly updated for continuation lines, breaking the distance checks used to determine if lists needed blank lines

The issues are fixed since 0.0.192.
